core/shared/src/main/scala/fs2/Task.scala

```diff
@@ -342,6 +342,9 @@ object Task extends TaskPlatform with TaskInstances {
     def set(t: Task[A]): Task[Unit] =
       Task.delay { S { t.unsafeRunAsync { r => actor ! Msg.Set(r) } }}
 
+    def setSync(t: Task[A]): Task[Unit] =
+      Task.delay { t.unsafeRunAsync { r => actor ! Msg.Set(r) }}
```

Comment:
Note that this is still asynchronous -- once the `Msg.Set(r)` message has veen sent to the actor, the resultant `Task[Unit]` is completed. We need to use `Task.async` and pass the callback along to the actor, letting the actor invoke the callback only after the `Set` has happened.